### PR TITLE
MaximGrdv

### PR DESCRIPTION
**Pull request by MaxGrdv**
added main class Server

### CODE REVIEW

File: Project6/src/main/java/org/nnstu6/server/core/Server.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    }
+
+    /**
+     * Method, that starts server
+     */
+    @Override
+    public void launchServer() {
+        try {
+            object = new Server();
```
Comment (SuppieRK) [inline Project6/src/main/java/org/nnstu6/server/core/Server.java]: Wtf? Why are you doing this, when you have `this`?

File: Project6/src/main/java/org/nnstu6/server/core/Server.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import java.util.UUID;
+
+/**
+ * Main class Server
+ *
+ * @author Max Gordeev
+ */
+public class Server extends AbstractServer implements ServerPerformance {
+    private Map<String, ClientIF> clients;
+    private Server object;
```
Comment (SuppieRK) [inline Project6/src/main/java/org/nnstu6/server/core/Server.java]: What is the point of `object`? Why `clients` map is not `final`?

File: Project6/src/main/java/org/nnstu6/server/core/Server.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+     * @return list of messages
+     */
+    @Override
+    public List<Messages> getMessages(UUID key, Timestamp time) throws RemoteException {
+
+        try {
+            return Database.getInstance().getMessages(creatingNewDialog(), time.getTime());
+        } catch (SQLException e) {
+            log.error(e);
+            throw new RemoteException();
```
Comment (SuppieRK) [inline Project6/src/main/java/org/nnstu6/server/core/Server.java]: Require correct Exception rethrowing (pass SQLException as a parameter to the exception constructor) or do this in more fail-safe way (return empty collection instead)

File: Project6/src/main/java/org/nnstu6/server/core/Server.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+     * Method that allows you to send message
+     *
+     * @param key     special key of recipient
+     * @param message content of message
+     * @param sender  who send the  message
+     * @param time    the moment when  you sent  new message
+     * @return -true, if message was sent successfully
+     */
+    @Override
+    public boolean sendMessage(UUID key, String message, String sender, Timestamp time) throws RemoteException {
```
Comment (SuppieRK) [inline Project6/src/main/java/org/nnstu6/server/core/Server.java]: Please, point out where does this methods throws `RemoteException`.

File: Project6/src/main/java/org/nnstu6/server/core/Server.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+
+    /**
+     * Method that allows you to create a new dialogue/conversation
+     *
+     * @return special key of dialog
+     */
+    @Override
+    public UUID creatingNewDialog() throws RemoteException {
+        return UUID.randomUUID();
```
Comment (SuppieRK) [inline Project6/src/main/java/org/nnstu6/server/core/Server.java]: I suppose that the implementation of this method is not finished?

File: Project6/src/main/java/org/nnstu6/server/core/Server.java [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+     * @param client   remote object
+     * @return true, if check-in was successful
+     */
+    @Override
+    public boolean authorization(String login, UUID password, boolean register, ClientIF client) throws RemoteException {
+
+        this.clients.put(login, client);
+        if (register) {
+
+            return Database.getInstance().authorizeUser(login, password);
```
Comment (SuppieRK) [inline Project6/src/main/java/org/nnstu6/server/core/Server.java]: Question - why you didn't store `Database` instance as non-static non-final member of class?

File: Project6/src/main/java/org/nnstu6/server/core/Server.java (hunk 4 above)
Comment (MaxGrdv) [inline Project6/src/main/java/org/nnstu6/server/core/Server.java]: That is? Explain,please

File: Project6/src/main/java/org/nnstu6/server/core/Server.java (hunk 5 above)
Comment (MaxGrdv) [inline Project6/src/main/java/org/nnstu6/server/core/Server.java]: Yeah

Comment (MaxGrdv) [inline Project6/src/main/java/org/nnstu6/server/core/Server.java]: will be added later

File: Project6/src/main/java/org/nnstu6/server/core/Server.java (hunk 4 above)
Comment (SuppieRK) [inline Project6/src/main/java/org/nnstu6/server/core/Server.java]: Where does this method throws `RemoteException`? Point a line there.

File: Project6/src/main/java/org/nnstu6/server/core/Server.java (hunk 6 above)
Comment (MaxGrdv) [inline Project6/src/main/java/org/nnstu6/server/core/Server.java]: Like this? -  private volatile Database instance;     And   return instance.authorizeUser(login, password);

File: Project6/src/main/java/org/nnstu6/server/core/Server.java (hunk 1 above)
Comment (MaxGrdv) [inline Project6/src/main/java/org/nnstu6/server/core/Server.java]: Like this?  -   UnicastRemoteObject.unexportObject(this, true);

File: Project6/src/main/java/org/nnstu6/server/core/Server.java (hunk 4 above)
Comment (MaxGrdv) [inline Project6/src/main/java/org/nnstu6/server/core/Server.java]: It is a method of the remote interface. 'Remote' provides only 'throws RemoteException'

Comment (MaxGrdv) [inline Project6/src/main/java/org/nnstu6/server/core/Server.java]: it's method*

**Pull request closed without merging**